napari/layers/points/points.py

```diff
@@ -595,8 +595,8 @@ def _extent_data(self) -> np.ndarray:
         if len(self.data) == 0:
             extrema = np.full((2, self.ndim), np.nan)
         else:
-            maxs = np.max(self.data, axis=0)
-            mins = np.min(self.data, axis=0)
+            maxs = np.max(self.data + self.size / 2, axis=0)
+            mins = np.min(self.data - self.size / 2, axis=0)
```

Comment:
Nice, I was tempted to make this change when looking #3444, but didn't want to introduce other new things in that PR. This seems consistent with incorporating the full pixel widths in the extent for image layers.